gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -554,7 +555,41 @@ CompilePatternCheckExpr::visit (HIR::SlicePattern &pattern)
     case TyTy::TypeKind::SLICE:
       rust_sorry_at (
 	pattern.get_locus (),
-	"SlicePattern matching against slices are not yet supported");
+	"SlicePattern matching against non-ref slices are not yet supported");
+      break;
+    case TyTy::TypeKind::REF:
+      {
+	rust_assert (RS_DST_FLAG_P (TREE_TYPE (match_scrutinee_expr)));
+	tree data_field
+	  = Backend::struct_field_expression (match_scrutinee_expr, 0,
+					      pattern.get_locus ());
+	tree size_field
+	  = Backend::struct_field_expression (match_scrutinee_expr, 1,
+					      pattern.get_locus ());
+
+	// First compare the size
+	check_expr = Backend::comparison_expression (
+	  ComparisonOperator::EQUAL, size_field,
+	  build_int_cst (size_type_node, pattern.get_items ().size ()),
+	  pattern.get_locus ());
+
+	// Then compare each element in the slice pattern
+	tree derefed = indirect_expression (data_field, pattern.get_locus ());
```

Comment:
I think we dont need the indirect_expression if you use the slice_index_expression function with the changes i suggested.